tenant-partitioning-config-service-impl/src/main/java/org/hypertrace/tenantpartitioning/config/service/TenantPartitioningConfigServiceImpl.java

```diff
@@ -0,0 +1,140 @@
+package org.hypertrace.tenantpartitioning.config.service;
+
+import com.google.inject.Inject;
+import com.typesafe.config.Config;
+import io.grpc.stub.StreamObserver;
+import java.util.List;
+import java.util.stream.Collectors;
+import lombok.extern.slf4j.Slf4j;
+import org.hypertrace.tenantpartitioning.config.service.store.TenantPartitionConfigStore;
+import org.hypertrace.tenantpartitioning.config.service.store.TenantPartitionGroupConfigKey;
+import org.hypertrace.tenantpartitioning.config.service.v1.*;
+
+@Slf4j
+public class TenantPartitioningConfigServiceImpl
+    extends TenantPartitioningConfigServiceGrpc.TenantPartitioningConfigServiceImplBase {
+
+  public static final String GENERIC_CONFIG_SERVICE = "generic.config.service";
+
+  private final TenantPartitionConfigStore tenantIsolationConfigStore;
+  private final TenantPartitioningConfigServiceRequestValidator validator;
+
+  @Inject
+  public TenantPartitioningConfigServiceImpl(
+      Config config,
+      TenantPartitionConfigStore tenantIsolationConfigStore,
+      TenantPartitioningConfigServiceRequestValidator validator) {
+    this.tenantIsolationConfigStore = tenantIsolationConfigStore;
+    this.validator = validator;
+    Config cfg = config.getConfig(GENERIC_CONFIG_SERVICE);
+    // this.tenantIsolationConfigStore = TenantPartitionConfigStoreProvider.getDocumentStore(cfg);
+  }
+
+  public void getTenantPartitionGroupConfigs(
+      GetTenantPartitionGroupConfigsRequest request,
+      StreamObserver<GetTenantPartitionGroupConfigsResponse> responseObserver) {
+    try {
+      validator.validateOrThrow(request);
+      List<TenantPartitionGroupConfig> configDTOs = tenantIsolationConfigStore.getAllGroupConfigs();
```

Comment:
these aren't DTOs any more - please fix naming. Also we seem to stream and map these into the same type - can that be removed?